src/Microsoft.Extensions.Configuration.Json/JsonConfigurationFileParser.cs

```diff
@@ -95,7 +95,7 @@ private void VisitPrimitive(JToken data)
             {
                 throw new FormatException(Resources.FormatError_KeyIsDuplicated(key));
             }
-            _data[key] = data.ToString();
+            _data[key] = data.Value<string>();
```

Comment:
Do we have good test coverage for all the node types mentioned in the switch statement? We need to be sure that moving to this API doesn't break the output for any if them.